### PR TITLE
feat: Hub captures exception

### PR DESCRIPTION
**Pull request by bruno-garcia**
* Hub catches and logs exception in callbacks and calls to dependencies
* When the Hub is disabled, method calls are no-op

### CODE REVIEW

File: sentry-core/src/test/java/io/sentry/core/HubTest.kt [hunk 1]
```diff
@@ -31,12 +38,42 @@ class HubTest {
...
-        assertEquals("transaction", scope.transaction)
-        assertEquals("fingerprint", scope.fingerprint[0])
+        assertEquals("transaction", clone.transaction)
+        assertEquals("fingerprint", clone.fingerprint.first())
         assertEquals("tags", clone.tags["tags"])
         assertEquals("a@a.com", clone.user.email)
     }
 
+    @Test
+    @Ignore("it's a shallow copy and we need a deep-copy") // TODO: https://www.baeldung.com/java-deep-copy
```
Comment (marandaneto) [inline sentry-core/src/test/java/io/sentry/core/HubTest.kt]: @bruno-garcia all the clones should be shallow or deep-copy? right now it returns a new instance of the object but it's fields are still the same.

File: sentry-core/src/main/java/io/sentry/core/Hub.java [hunk 2]
```diff
@@ -52,165 +54,310 @@ public boolean isEnabled() {
...
-      if (stack.size() != 1) {
-        stack.pop();
+    if (!isEnabled()) {
+      logIfNotNull(
+          options.getLogger(),
+          SentryLevel.WARNING,
+          "Instance is disabled and this 'popScope' call is a no-op.");
+    } else {
+      // Don't drop the root scope
+      synchronized (stack) { // TODO: is it necessary? we should never sync a concurrent object
```
Comment (marandaneto) [inline sentry-core/src/main/java/io/sentry/core/Hub.java]: @bruno-garcia  do we need that? `stack` is already thread-safe

File: sentry-core/src/test/java/io/sentry/core/HubTest.kt (hunk 1 above)
Comment (bruno-garcia) [reply sentry-core/src/test/java/io/sentry/core/HubTest.kt]: I want to say shallow is OK but for Scope data, if the value is not immutable, copying the reference would be a problem.

File: sentry-core/src/main/java/io/sentry/core/Hub.java (hunk 2 above)
Comment (bruno-garcia) [reply sentry-core/src/main/java/io/sentry/core/Hub.java]: There's a race here:

```
        if (stack.size() != 1) {
          stack.pop();
```

File: sentry-core/src/test/java/io/sentry/core/HubTest.kt (hunk 1 above)
Comment (marandaneto) [reply sentry-core/src/test/java/io/sentry/core/HubTest.kt]: @bruno-garcia ok so we already have a task on asana for it, default implementation of cloning on Java is shallow and it's also faulty if we make the fields `final`

**Pull request merged**